src/coord/timestamp.rs

```diff
@@ -574,7 +577,15 @@ impl Timestamper {
                         match cons {
                             Consistency::RealTime => {
                                 info!("Timestamping Source {} with Real Time Consistency", id);
-                                let last_offset = self.rt_recover_source(id);
+                                let start_offset = match sc {
+                                    ExternalSourceConnector::Kafka(KafkaSourceConnector {
+                                        start_offset,
+                                        ..
+                                    }) => start_offset,
+                                    _ => 0,
+                                };
+                                let last_offset =
+                                    std::cmp::max(start_offset, self.rt_recover_source(id));
```

Comment:
What happens when you shutdown Materialize and recover the views from the catalog? Is start_offset also persisted, or do you then recover from 0?

Reply:
`start_offset` should be persisted because it comes from the `WITH` option for a source, which I believe are persisted, but I will test that